--- ddclient.bak	2003-11-05 10:42:33.000000000 +1000
+++ ddclient	2003-11-05 10:52:55.000000000 +1000
@@ -20,7 +20,7 @@
 $program  =~ s/d$//;
 my $now       = time;
 my $hostname  = hostname();
-my $etc       = ($program =~ /test/i) ? './'   : '/etc/';
+my $etc       = ($program =~ /test/i) ? './'   : '/etc/ddclient/';
 my $savedir   = ($program =~ /test/i) ? 'URL/' : '/tmp/';
 my $msgs      = '';
 my $last_msgs = '';
