--- ddclient.bak	2003-11-05 10:42:33.000000000 +1000
+++ ddclient	2003-11-05 10:48:12.000000000 +1000
@@ -2629,7 +2629,7 @@
 		my $h = pop @hosts;
 	    
 		$config{$h}{'status'} = $status;
-		if ($status eq 'ok') {
+		if ($status eq 'ok' || $status eq 'nochange') {
 		    $config{$h}{'ip'}     = $ip;
 		    $config{$h}{'mtime'}  = $now;
 		    success("updating %s: %s: IP address set to %s", $h, $status, $ip);
