diff --git a/include/git2/version.h b/include/git2/version.h
index 6837e90..7572da1 100644
--- a/include/git2/version.h
+++ b/include/git2/version.h
@@ -7,12 +7,12 @@
 #ifndef INCLUDE_git_version_h__
 #define INCLUDE_git_version_h__
 
-#define LIBGIT2_VERSION "0.23.0"
+#define LIBGIT2_VERSION "0.24.0-rc1"
 #define LIBGIT2_VER_MAJOR 0
-#define LIBGIT2_VER_MINOR 23
+#define LIBGIT2_VER_MINOR 24
 #define LIBGIT2_VER_REVISION 0
 #define LIBGIT2_VER_PATCH 0
 
-#define LIBGIT2_SOVERSION 23
+#define LIBGIT2_SOVERSION 24
 
 #endif
