diff --git a/config.yml.example b/config.yml.example
index cf6c91b..7e144fe 100644
--- a/config.yml.example
+++ b/config.yml.example
@@ -25,15 +25,15 @@ http_settings:
   self_signed_cert: false
 
 # File used as authorized_keys for gitlab user
-auth_file: "/home/git/.ssh/authorized_keys"
+auth_file: "/var/lib/git/.ssh/authorized_keys"
 
 # File that contains the secret key for verifying access to GitLab.
-# Default is .gitlab_shell_secret in the gitlab-shell directory.
-# secret_file: "/home/git/gitlab-shell/.gitlab_shell_secret"
+# Default is /opt/gitlab/.gitlab_shell_secret.
+# secret_file: "/opt/gitlab/.gitlab_shell_secret"
 
 # Parent directory for global custom hook directories (pre-receive.d, update.d, post-receive.d)
 # Default is hooks in the gitlab-shell directory.
-# custom_hooks_dir: "/home/git/gitlab-shell/hooks"
+# custom_hooks_dir: "/usr/share/gitlab-shell/hooks"
 
 # Redis settings used for pushing commit notices to gitlab
 redis:
@@ -55,7 +55,7 @@ redis:
 
 # Log file.
 # Default is gitlab-shell.log in the root directory.
-# log_file: "/home/git/gitlab-shell/gitlab-shell.log"
+log_file: "/var/log/gitlab/gitlab-shell.log"
 
 # Log level. INFO by default
 log_level: INFO
diff --git a/lib/gitlab_config.rb b/lib/gitlab_config.rb
index a51a32c..7763802 100644
--- a/lib/gitlab_config.rb
+++ b/lib/gitlab_config.rb
@@ -4,11 +4,11 @@ class GitlabConfig
   attr_reader :config
 
   def initialize
-    @config = YAML.load_file(File.join(ROOT_PATH, 'config.yml'))
+    @config = YAML.load_file('/etc/gitlab-shell.yml')
   end
 
   def home
-    ENV['HOME']
+    "/var/lib/git"
   end
 
   def auth_file
@@ -16,7 +16,7 @@ class GitlabConfig
   end
 
   def secret_file
-    @config['secret_file'] ||= File.join(ROOT_PATH, '.gitlab_shell_secret')
+    @config['secret_file'] ||= "/opt/gitlab/.gitlab_shell_secret"
   end
 
   # Pass a default value because this is called from a repo's context; in which
@@ -43,7 +43,7 @@ class GitlabConfig
   end
 
   def log_file
-    @config['log_file'] ||= File.join(ROOT_PATH, 'gitlab-shell.log')
+    @config['log_file'] ||= "/var/log/gitlab/gitlab-shell.log"
   end
 
   def log_level
