--- a/lib/gitlab_config.rb
+++ b/lib/gitlab_config.rb
@@ -53,7 +53,7 @@ class GitlabConfig
       if redis.has_key?("socket")
         %W(#{redis['bin']} -s #{redis['socket']})
       else
-        %W(#{redis['bin']} -h #{redis['host']} -p #{redis['port']})
+        %W(#{redis['bin']} -h #{redis['host']} -p #{redis['port']} -n #{redis['db']})
       end
     end
   end
--- a/config.yml.example
+++ b/config.yml.example
@@ -25,6 +25,7 @@ redis:
   bin: /usr/bin/redis-cli
   host: 127.0.0.1
   port: 6379
+  # db: 0
   # socket: /tmp/redis.socket # Only define this if you want to use sockets
   namespace: resque:gitlab
 
