diff --git a/src/gitlab.com/gitlab-org/gitlab-ci-multi-runner/Makefile b/src/gitlab.com/gitlab-org/gitlab-ci-multi-runner/Makefile
index af9470e..099df3e 100644
--- a/src/gitlab.com/gitlab-org/gitlab-ci-multi-runner/Makefile
+++ b/src/gitlab.com/gitlab-org/gitlab-ci-multi-runner/Makefile
@@ -1,30 +1,21 @@
 NAME ?= gitlab-ci-multi-runner
 PACKAGE_NAME ?= $(NAME)
 PACKAGE_CONFLICT ?= $(PACKAGE_NAME)-beta
-VERSION := $(shell ./ci/version)
-REVISION := $(shell git rev-parse --short HEAD || echo unknown)
-BRANCH := $(shell git show-ref | grep "$(REVISION)" | grep -v HEAD | awk '{print $$2}' | sed 's|refs/remotes/origin/||' | sed 's|refs/heads/||' | sort | head -n 1)
+VERSION := @@VERSION@@
+REVISION := @@REVISION@@
+BRANCH := @@BRANCH@@
 BUILT := $(shell date +%Y-%m-%dT%H:%M:%S%:z)
 
-LATEST_STABLE_TAG := $(shell git -c versionsort.prereleaseSuffix="-rc" -c versionsort.prereleaseSuffix="-RC" tag -l "v*.*.*" --sort=-v:refname | awk '!/rc/' | head -n 1)
-export IS_LATEST :=
-ifeq ($(shell git describe --exact-match --match $(LATEST_STABLE_TAG) >/dev/null 2>&1; echo $$?), 0)
 export IS_LATEST := true
-endif
 
 PACKAGE_CLOUD ?= ayufan/gitlab-ci-multi-runner
 PACKAGE_CLOUD_URL ?= https://packagecloud.io/
-BUILD_PLATFORMS ?= -os '!netbsd' -os '!openbsd'
+BUILD_PLATFORMS ?= -osarch '@@OSARCH@@'
 S3_UPLOAD_PATH ?= master
-DEB_PLATFORMS ?= debian/wheezy debian/jessie debian/stretch debian/buster \
-    ubuntu/precise ubuntu/trusty ubuntu/utopic ubuntu/vivid ubuntu/wily ubuntu/xenial \
-    raspbian/wheezy raspbian/jessie raspbian/stretch raspbian/buster \
-    linuxmint/petra linuxmint/qiana linuxmint/rebecca linuxmint/rafaela linuxmint/rosa
-DEB_ARCHS ?= amd64 i386 armel armhf
-RPM_PLATFORMS ?= el/6 el/7 \
-    ol/6 ol/7 \
-    fedora/20 fedora/21 fedora/22 fedora/23
-RPM_ARCHS ?= x86_64 i686 arm armhf
+DEB_PLATFORMS ?= debian/wheezy
+DEB_ARCHS ?= amd64
+RPM_PLATFORMS ?= fedora/23
+RPM_ARCHS ?= x86_64
 COMMON_PACKAGE_NAMESPACE=$(shell go list ./common)
 
 # Packages in vendor/ are included in ./...
@@ -77,7 +68,7 @@ verify: fmt vet lint complexity test
 deps:
 	# Installing dependencies...
 	go get -u github.com/golang/lint/golint
-	go get github.com/mitchellh/gox
+	go get github.com/fritteli/gox
 	go get golang.org/x/tools/cmd/cover
 	go get github.com/fzipp/gocyclo
 	go get -u github.com/jteeuwen/go-bindata/...
@@ -108,7 +99,7 @@ else
 	$(warning WARNING: and remove out/docker/prebuilt-x86_64.tar.xz)
 	$(warning =============================================)
 	curl -o out/docker/prebuilt-x86_64.tar.xz \
-		https://gitlab-ci-multi-runner-downloads.s3.amazonaws.com/master/docker/prebuilt-x86_64.tar.xz
+		https://gitlab-ci-multi-runner-downloads.s3.amazonaws.com/$(VERSION)/docker/prebuilt-x86_64.tar.xz
 endif
 
 out/docker/prebuilt-arm.tar.xz: $(GO_FILES)
@@ -136,7 +127,7 @@ else
 	$(warning WARNING: and remove out/docker/prebuilt-arm.tar.xz)
 	$(warning =============================================)
 	curl -o out/docker/prebuilt-arm.tar.xz \
-		https://gitlab-ci-multi-runner-downloads.s3.amazonaws.com/master/docker/prebuilt-arm.tar.xz
+		https://gitlab-ci-multi-runner-downloads.s3.amazonaws.com/$(VERSION)/docker/prebuilt-arm.tar.xz
 endif
 
 executors/docker/bindata.go: out/docker/prebuilt-x86_64.tar.xz out/docker/prebuilt-arm.tar.xz
