diff --git a/config.toml.example b/config.toml.example
index 3a987f9..801d192 100644
--- a/config.toml.example
+++ b/config.toml.example
@@ -1,6 +1,6 @@
 # Example Gitaly configuration file
 
-socket_path = "/home/git/gitlab/tmp/sockets/private/gitaly.socket"
+socket_path = "/run/gitlab/gitaly.socket"
 
 # # Optional: listen on a TCP socket. This is insecure (no authentication)
 # listen_addr = "localhost:9999"
@@ -12,7 +12,7 @@ socket_path = "/home/git/gitlab/tmp/sockets/private/gitaly.socket"
 
 [[storage]]
 name = "default"
-path = "/home/git/repositories"
+path = "/var/lib/git/repositories"
 
 # # You can optionally configure more storages for this Gitaly instance to serve up
 #
