diff --git a/Makefile b/Makefile
index c2c8c59..419f4d8 100644
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,8 @@
-PREFIX=/usr/local
-VERSION=$(shell git describe)-$(shell date -u +%Y%m%d.%H%M%S)
+PREFIX=${DESTDIR}/usr
+VERSION=0.5.4-$(shell date -u +%Y%m%d.%H%M%S)
 
 gitlab-workhorse: $(wildcard *.go)
-	go build -ldflags "-X main.Version ${VERSION}" -o gitlab-workhorse
+	go build -ldflags "-X main.Version=${VERSION}" -o gitlab-workhorse
 
 install: gitlab-workhorse
 	install gitlab-workhorse ${PREFIX}/bin/
