diff --git a/src/gitlab.com/gitlab-org/gitlab-pages/Makefile.internal.mk b/src/gitlab.com/gitlab-org/gitlab-pages/Makefile.internal.mk
index 309083f..b9f3bc1 100644
--- a/src/gitlab.com/gitlab-org/gitlab-pages/Makefile.internal.mk
+++ b/src/gitlab.com/gitlab-org/gitlab-pages/Makefile.internal.mk
@@ -1,12 +1,6 @@
-REVISION := $(shell git rev-parse --short HEAD || echo unknown)
-LAST_TAG := $(shell git describe --tags --abbrev=0)
-COMMITS := $(shell echo `git log --oneline $(LAST_TAG)..HEAD | wc -l`)
+REVISION := @@REVISION@@
 VERSION := $(shell cat VERSION)
 
-ifneq (v$(VERSION),$(LAST_TAG))
-	VERSION := $(shell echo $(VERSION)~beta.$(COMMITS).g$(REVISION))
-endif
-
 VERSION_FLAGS := -ldflags='-X "main.Version=$(VERSION)" -X "main.REVISION=$(REVISION)"'
 
 # cd into the GOPATH to workaround ./... not following symlinks
