diff --exclude-from=/home/dang/.diffrc -u -ruN evince-0.5.3.orig/data/evince.desktop.in.in evince-0.5.3/data/evince.desktop.in.in
--- evince-0.5.3.orig/data/evince.desktop.in.in	2006-05-06 17:52:05.000000000 -0400
+++ evince-0.5.3/data/evince.desktop.in.in	2006-05-16 11:35:54.000000000 -0400
@@ -9,7 +9,6 @@
 Terminal=false
 Type=Application
 Icon=evince.png
-NoDisplay=true
 X-GNOME-DocPath=
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=evince
