diff --git a/Makefile.am b/Makefile.am
index 95d0807..105222b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -9,7 +9,7 @@ GMETAD_SUBDIR =
 endif
 
 EXTRA_SUBDIRS = "gmetad-python contrib solaris"
-INCLUDES = @APR_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @RPC_CFLAGS@
 
 AUTOMAKE_OPTIONS = dist-bzip2 dist-tarZ dist-zip
 ACLOCAL_AMFLAGS = -I m4
diff --git a/configure.ac b/configure.ac
index fe7983b..597f0ca 100644
--- a/configure.ac
+++ b/configure.ac
@@ -606,13 +606,20 @@ AC_CHECK_HEADER([sys/mount.h], [], [],
 #endif
 ])
 
-AC_CHECK_HEADERS(rpc/types.h)
-AC_CHECK_HEADER([rpc/xdr.h], [],
-   [AC_MSG_ERROR([your system is missing the Sun RPC (ONC/RPC) libraries])],
-[#ifdef HAVE_RPC_TYPES_H
-# include <rpc/types.h>
-#endif
-])
+dnl ##################################################################
+dnl Checks for TIRPC
+PKG_CHECK_MODULES([TIRPC],[libtirpc])
+dnl LDFLAGS="$LDFLAGS $LIBS $APR_LIBS"
+LDFLAGS="$LDFLAGS $TIRPC_LIBS"
+RPC_CFLAGS="$TIRPC_CFLAGS"
+
+dnl PKG_CHECK_MODULES([TIRPC],
+dnl                   [libtirpc],
+dnl                   [RPC_CFLAGS=$TIRPC_CFLAGS; RPC_LIBS=$TIRPC_LIBS;],
+dnl                   [AC_MSG_ERROR([libtirpc requested, but library not found.])]
+dnl )
+nl AC_SUBST(RPC_CFLAGS)
+dnl AC_SUBST(RPC_LIBS)
 
 dnl ##################################################################
 dnl Checks for typedefs.
diff --git a/gmetad/Makefile.am b/gmetad/Makefile.am
index 0a4e360..874d448 100644
--- a/gmetad/Makefile.am
+++ b/gmetad/Makefile.am
@@ -11,7 +11,7 @@ GLDADD =
 GLDFLAGS = 
 endif
 
-INCLUDES = @APR_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @RPC_CFLAGS@
 AM_CFLAGS = -I$(top_builddir)/lib -I$(top_builddir)/gmond -I$(top_builddir)/libmetrics -I$(top_builddir)/include $(GCFLAGS) @PROTOBUF_C_CFLAGS@
 
 sbin_PROGRAMS = gmetad
diff --git a/gmetric/Makefile.am b/gmetric/Makefile.am
index bc7807c..d9bf585 100644
--- a/gmetric/Makefile.am
+++ b/gmetric/Makefile.am
@@ -10,7 +10,7 @@ GLDADD =
 GLDFLAGS = 
 endif
 
-INCLUDES = @APR_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @RPC_CFLAGS@
 
 AM_CFLAGS = -I../lib -I../include $(GCFLAGS)
 
diff --git a/gmond/Makefile.am b/gmond/Makefile.am
index 256fee3..d6f5f07 100644
--- a/gmond/Makefile.am
+++ b/gmond/Makefile.am
@@ -44,7 +44,7 @@ if BUILD_SFLOW
 GCFLAGS += -DSFLOW
 endif
 
-INCLUDES = @APR_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @RPC_CFLAGS@
 AM_CFLAGS = -I$(top_builddir)/lib -I$(top_builddir)/include/ -I$(top_builddir)/libmetrics $(GCFLAGS)
 
 sbin_PROGRAMS = gmond
diff --git a/gmond/modules/cpu/Makefile.am b/gmond/modules/cpu/Makefile.am
index fd7457e..99ddb83 100644
--- a/gmond/modules/cpu/Makefile.am
+++ b/gmond/modules/cpu/Makefile.am
@@ -25,4 +25,4 @@ modload_la_LIBADD = $(top_builddir)/libmetrics/libmetrics.la
 
 endif
 
-INCLUDES = @APR_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @RPC_CFLAGS@
diff --git a/gmond/modules/disk/Makefile.am b/gmond/modules/disk/Makefile.am
index 0c78689..2f5a2f0 100644
--- a/gmond/modules/disk/Makefile.am
+++ b/gmond/modules/disk/Makefile.am
@@ -11,4 +11,4 @@ moddisk_la_LDFLAGS = -module -avoid-version
 moddisk_la_LIBADD = $(top_builddir)/libmetrics/libmetrics.la
 endif
 
-INCLUDES = @APR_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @RPC_CFLAGS@
diff --git a/gmond/modules/example/Makefile.am b/gmond/modules/example/Makefile.am
index e3bff6a..18b4e3b 100644
--- a/gmond/modules/example/Makefile.am
+++ b/gmond/modules/example/Makefile.am
@@ -11,7 +11,7 @@ modexample_la_LDFLAGS = -module -avoid-version
 EXTRA_DIST = ../conf.d/example.conf
 endif
 
-INCLUDES = @APR_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @RPC_CFLAGS@
 
 install:
 	@echo
diff --git a/gmond/modules/memory/Makefile.am b/gmond/modules/memory/Makefile.am
index 82336db..06ad430 100644
--- a/gmond/modules/memory/Makefile.am
+++ b/gmond/modules/memory/Makefile.am
@@ -12,4 +12,4 @@ modmem_la_LDFLAGS = -module -avoid-version
 modmem_la_LIBADD = $(top_builddir)/libmetrics/libmetrics.la
 endif
 
-INCLUDES = @APR_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @RPC_CFLAGS@
diff --git a/gmond/modules/network/Makefile.am b/gmond/modules/network/Makefile.am
index 72a4faf..c370d88 100644
--- a/gmond/modules/network/Makefile.am
+++ b/gmond/modules/network/Makefile.am
@@ -12,4 +12,4 @@ modnet_la_LDFLAGS = -module -avoid-version
 modnet_la_LIBADD = $(top_builddir)/libmetrics/libmetrics.la
 endif
 
-INCLUDES = @APR_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @RPC_CFLAGS@
diff --git a/gmond/modules/perl/Makefile.am b/gmond/modules/perl/Makefile.am
index b7b095c..f1d0fb9 100644
--- a/gmond/modules/perl/Makefile.am
+++ b/gmond/modules/perl/Makefile.am
@@ -28,4 +28,4 @@ install-exec-hook:	../conf.d/modperl.conf README
 	mkdir -p $(DESTDIR)$(sysconfdir)/conf.d && \
 	  $(INSTALL_DATA) ../conf.d/modperl.conf $(DESTDIR)$(sysconfdir)/conf.d/modperl.conf
 
-INCLUDES = @APR_INCLUDES@ @PERL_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @PERL_INCLUDES@ @RPC_CFLAGS@
diff --git a/gmond/modules/python/Makefile.am b/gmond/modules/python/Makefile.am
index f535b07..c8fcac0 100644
--- a/gmond/modules/python/Makefile.am
+++ b/gmond/modules/python/Makefile.am
@@ -27,4 +27,4 @@ install-exec-hook:	../conf.d/modpython.conf README
 	mkdir -p $(DESTDIR)$(sysconfdir)/conf.d && \
 	  $(INSTALL_DATA) ../conf.d/modpython.conf $(DESTDIR)$(sysconfdir)/conf.d/modpython.conf
 
-INCLUDES = @APR_INCLUDES@ @PYTHON_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @PYTHON_INCLUDES@ @RPC_CFLAGS@
diff --git a/gmond/modules/status/Makefile.am b/gmond/modules/status/Makefile.am
index cd84069..53968fd 100644
--- a/gmond/modules/status/Makefile.am
+++ b/gmond/modules/status/Makefile.am
@@ -11,4 +11,4 @@ modgstatus_la_LDFLAGS = -module -avoid-version
 EXTRA_DIST = ../conf.d/modgstatus.conf
 endif
 
-INCLUDES = @APR_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @RPC_CFLAGS@
diff --git a/gmond/modules/system/Makefile.am b/gmond/modules/system/Makefile.am
index 5fc7fb3..566469e 100644
--- a/gmond/modules/system/Makefile.am
+++ b/gmond/modules/system/Makefile.am
@@ -16,4 +16,4 @@ modsys_la_LDFLAGS = -module -avoid-version
 modsys_la_LIBADD = $(top_builddir)/libmetrics/libmetrics.la
 endif
 
-INCLUDES = @APR_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @RPC_CFLAGS@
diff --git a/lib/Makefile.am b/lib/Makefile.am
index 935c514..6e0499f 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -14,7 +14,7 @@ if BUILD_SFLOW
 GCFLAGS += -DSFLOW
 endif
 
-INCLUDES = @APR_INCLUDES@
+INCLUDES = @APR_INCLUDES@ @RPC_CFLAGS@
 AM_CFLAGS = -I.. -I. -I$(top_builddir)/include/ $(GCFLAGS) -DSYSCONFDIR='"$(sysconfdir)"'
 
 include_HEADERS = gm_protocol.h
diff --git a/scripts/fixconfig b/scripts/fixconfig
index 5be7982..707f62d 100755
--- a/scripts/fixconfig
+++ b/scripts/fixconfig
@@ -35,7 +35,7 @@ fi
 
 # Additional variables set by configure
 CFLAGS="`echo '-g -O2 -fno-strict-aliasing -Wall -D_REENTRANT' | $AWK '{ gsub(/,/, "\\\\,"); print }'`"
-LDFLAGS="`echo ' -lpthread  -lapr-1  ' | $AWK '{ gsub(/,/, "\\\\,"); print }'`"
+LDFLAGS="`echo ' -lpthread  -lapr-1 -ltirpc  ' | $AWK '{ gsub(/,/, "\\\\,"); print }'`"
 LIBS="-ldl -lnsl -lnsl -lz -lpcre -lexpat -lconfuse -lpthread "
 VERSION="3.7.2"
 host_cpu="x86_64"
