--- common-orig/lib/modules/fglrx/build_mod/firegl_public.c	2007-08-01 16:58:40.000000000 -0500
+++ common/lib/modules/fglrx/build_mod/firegl_public.c	2007-08-01 16:57:11.000000000 -0500
@@ -144,12 +144,14 @@
 #define EXPORT_NO_SYMBOLS
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22)
 #ifdef __x86_64__
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,12)
 #include "linux/ioctl32.h"
 #else
 #include "asm/ioctl32.h"
 #endif
+#endif
 
 #ifdef __x86_64__
 #include "asm/compat.h"
