diff --git a/app/mailers/notify.rb b/app/mailers/notify.rb
index aec4c57..eec1480 100644
--- a/app/mailers/notify.rb
+++ b/app/mailers/notify.rb
@@ -16,7 +16,7 @@ class Notify < ActionMailer::Base
   default_url_options[:script_name] = Gitlab.config.gitlab.relative_url_root
 
   default from: Gitlab.config.gitlab.email_from
-  default reply_to: "noreply@#{Gitlab.config.gitlab.host}"
+  default reply_to: Gitlab.config.gitlab.email_reply_to
 
   # Just send email with 3 seconds delay
   def self.delay
diff --git a/config/gitlab.yml.example b/config/gitlab.yml.example
index c63e8cb..776d930 100644
--- a/config/gitlab.yml.example
+++ b/config/gitlab.yml.example
@@ -37,6 +37,9 @@ production: &base
     # Email address used in the "From" field in mails sent by GitLab
     email_from: gitlab@localhost
 
+    # Email address used in the "Reply-To" field in mails send by GitLab (default: no-reply@<host>)
+    email_reply_to: no-reply@localhost
+
     # Email address of your support contact (default: same as email_from)
     support_email: support@localhost
 
diff --git a/config/initializers/1_settings.rb b/config/initializers/1_settings.rb
index ea391ca..580d65b 100644
--- a/config/initializers/1_settings.rb
+++ b/config/initializers/1_settings.rb
@@ -82,6 +82,7 @@ Settings.gitlab['port']       ||= Settings.gitlab.https ? 443 : 80
 Settings.gitlab['relative_url_root'] ||= ENV['RAILS_RELATIVE_URL_ROOT'] || ''
 Settings.gitlab['protocol']   ||= Settings.gitlab.https ? "https" : "http"
 Settings.gitlab['email_from'] ||= "gitlab@#{Settings.gitlab.host}"
+Settings.gitlab['email_reply_to'] ||= "no-reply@#{Settings.gitlab.host}"
 Settings.gitlab['support_email']  ||= Settings.gitlab.email_from
 Settings.gitlab['url']        ||= Settings.send(:build_gitlab_url)
 Settings.gitlab['user']       ||= 'git'
