--- app/mailers/notify.rb.orig	2014-05-19 12:22:50.840000000 +0200
+++ app/mailers/notify.rb	2014-05-19 12:23:35.660000000 +0200
@@ -16,7 +16,7 @@
   default_url_options[:script_name] = Gitlab.config.gitlab.relative_url_root
 
   default from: Proc.new { default_sender_address.format }
-  default reply_to: "noreply@#{Gitlab.config.gitlab.host}"
+  default reply_to: Gitlab.config.gitlab.email_reply_to
 
   # Just send email with 2 seconds delay
   def self.delay
--- config/gitlab.yml.example.orig	2014-05-19 12:23:45.640000000 +0200
+++ config/gitlab.yml.example	2014-05-19 12:24:07.340000000 +0200
@@ -37,6 +37,9 @@
     # Email address used in the "From" field in mails sent by GitLab
     email_from: gitlab@localhost
 
+    # Email address used in the "Reply-To" field in mails send by GitLab (default: no-reply@<host>)
+    email_reply_to: no-reply@localhost
+
     # Email address of your support contact (default: same as email_from)
     support_email: support@localhost
 
--- config/initializers/1_settings.rb.orig	2014-05-19 12:24:14.520000000 +0200
+++ config/initializers/1_settings.rb	2014-05-19 12:24:44.010000000 +0200
@@ -78,6 +78,7 @@
 Settings.gitlab['relative_url_root'] ||= ENV['RAILS_RELATIVE_URL_ROOT'] || ''
 Settings.gitlab['protocol']   ||= Settings.gitlab.https ? "https" : "http"
 Settings.gitlab['email_from'] ||= "gitlab@#{Settings.gitlab.host}"
+Settings.gitlab['email_reply_to'] ||= "no-reply@#{Settings.gitlab.host}"
 Settings.gitlab['support_email']  ||= Settings.gitlab.email_from
 Settings.gitlab['url']        ||= Settings.send(:build_gitlab_url)
 Settings.gitlab['user']       ||= 'git'
