--- src/ptlib/common/jidctflt.cxx.orig	2008-09-24 09:13:06.000000000 -0400
+++ src/ptlib/common/jidctflt.cxx	2008-09-24 09:11:49.000000000 -0400
@@ -110,10 +110,10 @@
       "\tsar %2,%1\n"
       "\tsub $-128,%1\n"
       "\tcmovl %5,%1\n"	/* Use the sub to compare to 0 */
-      "\tcmpl %4,%1\n" 
+      "\tcmp %4,%1\n" 
       "\tcmovg %4,%1\n"
       : "=r"(x) 
-      : "0"(x), "Ir"(shift), "ir"(1UL<<(shift-1)), "r" (0xff), "r" (0)
+      : "0"((unsigned long)x), "c"((char)shift), "ir"(1UL<<(shift-1)), "r" (0xffUL), "r" (0UL)
       );
   return x;
 }
