diff --git a/cmd/mautrix-whatsapp/main.go b/cmd/mautrix-whatsapp/main.go
index a3e0665..647b65b 100644
--- a/cmd/mautrix-whatsapp/main.go
+++ b/cmd/mautrix-whatsapp/main.go
@@ -9,8 +9,8 @@ import (
 // Information to find out exactly which commit the bridge was built from.
 // These are filled at build time with the -X linker flag.
 var (
-	Tag       = "unknown"
-	Commit    = "unknown"
+	Tag       = "v0.2510.0"
+	Commit    = "9802479e6b67c71d04113cb1721848f0d37c236e"
 	BuildTime = "unknown"
 )
 
