diff --git a/config/database.yml.mysql b/config/database.yml.mysql
index 98c2abe..2896c50 100644
--- a/config/database.yml.mysql
+++ b/config/database.yml.mysql
@@ -3,58 +3,12 @@
 #
 production:
   adapter: mysql2
-  encoding: utf8
-  collation: utf8_general_ci
-  reconnect: false
-  database: gitlabhq_production
-  pool: 10
-  username: git
-  password: "secure password"
-  host: localhost
-  # socket: /tmp/mysql.sock
-
-#
-# Development specific
-#
-development:
-  adapter: mysql2
-  encoding: utf8
-  collation: utf8_general_ci
-  reconnect: false
-  database: gitlabhq_development
-  pool: 5
-  username: root
-  password: "secure password"
-  host: localhost
-  # socket: /tmp/mysql.sock
-
-#
-# Staging specific
-#
-staging:
-  adapter: mysql2
-  encoding: utf8
-  collation: utf8_general_ci
+  encoding: utf8mb4
+  collation: utf8mb4_general_ci
   reconnect: false
-  database: gitlabhq_staging
+  database: gitlab
   pool: 10
   username: git
   password: "secure password"
   host: localhost
-  # socket: /tmp/mysql.sock
-
-# Warning: The database defined as "test" will be erased and
-# re-generated from your development database when you run "rake".
-# Do not set this db to the same as development or production.
-test: &test
-  adapter: mysql2
-  encoding: utf8mb4
-  collation: utf8mb4_general_ci
-  reconnect: false
-  database: gitlabhq_test
-  pool: 5
-  username: root
-  password:
-  host: localhost
-  # socket: /tmp/mysql.sock
-  prepared_statements: false
+  # socket: /run/mysqld/mysqld.sock
diff --git a/config/database.yml.postgresql b/config/database.yml.postgresql
index baded68..5669617 100644
--- a/config/database.yml.postgresql
+++ b/config/database.yml.postgresql
@@ -4,45 +4,8 @@
 production:
   adapter: postgresql
   encoding: unicode
-  database: gitlabhq_production
+  database: gitlab
   pool: 10
   username: git
   password: "secure password"
   host: localhost
-
-#
-# Development specific
-#
-development:
-  adapter: postgresql
-  encoding: unicode
-  database: gitlabhq_development
-  pool: 5
-  username: postgres
-  password: "secure password"
-  host: localhost
-
-#
-# Staging specific
-#
-staging:
-  adapter: postgresql
-  encoding: unicode
-  database: gitlabhq_staging
-  pool: 10
-  username: git
-  password: "secure password"
-  host: localhost
-
-# Warning: The database defined as "test" will be erased and
-# re-generated from your development database when you run "rake".
-# Do not set this db to the same as development or production.
-test: &test
-  adapter: postgresql
-  encoding: unicode
-  database: gitlabhq_test
-  pool: 5
-  username: postgres
-  password:
-  host: localhost
-  prepared_statements: false
