diff --git a/lib/tasks/gitlab/check.rake b/lib/tasks/gitlab/check.rake
index 63c5e9b..2ffa8ed 100644
--- a/lib/tasks/gitlab/check.rake
+++ b/lib/tasks/gitlab/check.rake
@@ -28,7 +28,7 @@ namespace :gitlab do
         SystemCheck::App::UploadsPathPermissionCheck,
         SystemCheck::App::UploadsPathTmpPermissionCheck,
         SystemCheck::App::InitScriptExistsCheck,
-        SystemCheck::App::InitScriptUpToDateCheck,
+        # SystemCheck::App::InitScriptUpToDateCheck, Gentoo uses a different init script
         SystemCheck::App::ProjectsHaveNamespaceCheck,
         SystemCheck::App::RedisVersionCheck,
         SystemCheck::App::RubyVersionCheck,
@@ -272,13 +272,6 @@ namespace :gitlab do
         puts "yes".color(:green)
       else
         puts "no".color(:red)
-        try_fixing_it(
-          sudo_gitlab("RAILS_ENV=production bin/background_jobs start")
-        )
-        for_more_information(
-          see_installation_guide_section("Install Init Script"),
-          "see log/sidekiq.log for possible errors"
-        )
         fix_and_rerun
       end
     end
