diff --git a/lib/tasks/gitlab/check.rake b/lib/tasks/gitlab/check.rake
index 5f4a6bb..8f75d2b 100644
--- a/lib/tasks/gitlab/check.rake
+++ b/lib/tasks/gitlab/check.rake
@@ -24,7 +24,7 @@ namespace :gitlab do
       check_tmp_writable
       check_uploads
       check_init_script_exists
-      check_init_script_up_to_date
+      #check_init_script_up_to_date Gentoo uses a different init script
       check_projects_have_namespace
       check_redis_version
       check_ruby_version
@@ -590,13 +590,6 @@ namespace :gitlab do
         puts "yes".color(:green)
       else
         puts "no".color(:red)
-        try_fixing_it(
-          sudo_gitlab("RAILS_ENV=production bin/background_jobs start")
-        )
-        for_more_information(
-          see_installation_guide_section("Install Init Script"),
-          "see log/sidekiq.log for possible errors"
-        )
         fix_and_rerun
       end
     end
