diff --git a/config/environments/production.rb b/config/environments/production.rb
index a9d8ac4..256770a 100644
--- a/config/environments/production.rb
+++ b/config/environments/production.rb
@@ -69,11 +69,10 @@ Rails.application.configure do
   config.active_support.deprecation = :notify
 
   config.action_mailer.delivery_method = :sendmail
-  # Defaults to:
-  # # config.action_mailer.sendmail_settings = {
-  # #   location: '/usr/sbin/sendmail',
-  # #   arguments: '-i -t'
-  # # }
+  config.action_mailer.sendmail_settings = {
+    location: '/usr/sbin/sendmail',
+    arguments: '-i'
+  }
   config.action_mailer.perform_deliveries = true
   config.action_mailer.raise_delivery_errors = true
 
