diff --git a/Installer/Makefile/Makefile b/Installer/Makefile/Makefile
index 5528c7e..0dfa485 100644
--- a/Installer/Makefile/Makefile
+++ b/Installer/Makefile/Makefile
@@ -2,20 +2,14 @@ BUILD_TOOL=xbuild
 BUILD_ARGS=/property:Platform=Any\ CPU /property:Configuration=Release
 OUTPUT_ARGS=/property:OutputPath=bin/Release
 SOURCE_DIR=../../.
-PACKAGE_DIR=./pkg
+PACKAGE_DIR=${D}/.
 PKG_TARGET=$(PACKAGE_DIR)/usr/lib/duplicati
 LOCALIZATION_TOOL=mono $(SOURCE_DIR)/Duplicati/Localization/LocalizationTool.exe
 
 all: package
 
-Duplicati-v1.sln:
-	patch -d "$(SOURCE_DIR)" -i patch-1-remove-scheduler -o Duplicati-v1.sln Duplicati.sln
-
-Duplicati-v2.sln: Duplicati-v1.sln
-	patch -d "$(SOURCE_DIR)" -i patch-2-remove-unsigned-backends -o Duplicati-v2.sln Duplicati-v1.sln
-
-build: Duplicati-v2.sln
-	$(BUILD_TOOL) $(BUILD_ARGS) "$(SOURCE_DIR)/Duplicati-v2.sln"
+build:
+	$(BUILD_TOOL) $(BUILD_ARGS) "$(SOURCE_DIR)/Duplicati.sln"
 	$(BUILD_TOOL) $(BUILD_ARGS) $(OUTPUT_ARGS) "$(SOURCE_DIR)/Duplicati/Library/Backend/SSH/Duplicati.Library.Backend.SSH.csproj"
 	$(BUILD_TOOL) $(BUILD_ARGS) $(OUTPUT_ARGS) "$(SOURCE_DIR)/Duplicati/Library/Backend/TahoeLAFS/Duplicati.Library.Backend.TahoeLAFS.csproj"
 	$(BUILD_TOOL) $(BUILD_ARGS) "$(SOURCE_DIR)/BuildTools/LocalizationTool/LocalizationTool.sln"
@@ -31,10 +25,6 @@ translations: build
 	$(LOCALIZATION_TOOL) build
 
 package:
-	if [ -e "$(PACKAGE_DIR)" ]; then \
-		rm -rf "$(PACKAGE_DIR)"; \
-	fi
-	mkdir "$(PACKAGE_DIR)"
 	mkdir "$(PACKAGE_DIR)/usr"
 	mkdir "$(PACKAGE_DIR)/usr/lib"
 	mkdir "$(PACKAGE_DIR)/usr/bin"
@@ -51,7 +41,6 @@ package:
 	cp -R "$(SOURCE_DIR)/Duplicati/Localization/compiled/"* "$(PKG_TARGET)"
 
 	rm -rf "$(PKG_TARGET)/alphavss"
-	rm -rf "$(PKG_TARGET)/SQLite"
 	rm -rf "$(PKG_TARGET)/win-tools"
 	rm -rf "$(PKG_TARGET)/"*.mdb
 	rm -rf "$(PKG_TARGET)/AlphaFS.dll"
@@ -79,16 +68,6 @@ package:
 	
 	chmod +x "$(PACKAGE_DIR)/usr/bin/duplicati"
 	chmod +x "$(PACKAGE_DIR)/usr/bin/duplicati-commandline"
-	
-	if [ -e "duplicati.tar" ]; then \
-		rm -rf "duplicati.tar"; \
-	fi
-	if [ -e "duplicati.tar.gz" ]; then \
-		rm -rf "duplicati.tar.gz"; \
-	fi
-	
-	tar -C "$(PACKAGE_DIR)" -cf "duplicati.tar" .
-	gzip --best "duplicati.tar"
 
 clean:
 	rm -rf "$(SOURCE_DIR)/Duplicati-v1.sln"
