diff --git a/Duplicati/Library/Backend/FTP/Duplicati.Library.Backend.FTP.csproj b/Duplicati/Library/Backend/FTP/Duplicati.Library.Backend.FTP.csproj
index 2b4c161..10902eb 100644
--- a/Duplicati/Library/Backend/FTP/Duplicati.Library.Backend.FTP.csproj
+++ b/Duplicati/Library/Backend/FTP/Duplicati.Library.Backend.FTP.csproj
@@ -64,7 +64,7 @@
     <Compile Include="FTPUI.cs">
       <SubType>UserControl</SubType>
     </Compile>
-    <Compile Include="FTPUI.designer.cs">
+    <Compile Include="FTPUI.Designer.cs">
       <DependentUpon>FTPUI.cs</DependentUpon>
     </Compile>
     <Compile Include="Properties\AssemblyInfo.cs" />
diff --git a/Duplicati/Library/Backend/File/Duplicati.Library.Backend.File.csproj b/Duplicati/Library/Backend/File/Duplicati.Library.Backend.File.csproj
index 0b03e11..5e5ce7f 100644
--- a/Duplicati/Library/Backend/File/Duplicati.Library.Backend.File.csproj
+++ b/Duplicati/Library/Backend/File/Duplicati.Library.Backend.File.csproj
@@ -64,7 +64,7 @@
     <Compile Include="FileUI.cs">
       <SubType>UserControl</SubType>
     </Compile>
-    <Compile Include="FileUI.designer.cs">
+    <Compile Include="FileUI.Designer.cs">
       <DependentUpon>FileUI.cs</DependentUpon>
     </Compile>
     <Compile Include="Properties\AssemblyInfo.cs" />
diff --git a/Duplicati/Library/Backend/S3/Duplicati.Library.Backend.S3.csproj b/Duplicati/Library/Backend/S3/Duplicati.Library.Backend.S3.csproj
index ddc0f47..4dfadfc 100644
--- a/Duplicati/Library/Backend/S3/Duplicati.Library.Backend.S3.csproj
+++ b/Duplicati/Library/Backend/S3/Duplicati.Library.Backend.S3.csproj
@@ -77,7 +77,7 @@
     <Compile Include="S3UI.cs">
       <SubType>UserControl</SubType>
     </Compile>
-    <Compile Include="S3UI.designer.cs">
+    <Compile Include="S3UI.Designer.cs">
       <DependentUpon>S3UI.cs</DependentUpon>
     </Compile>
     <Compile Include="S3Wrapper.cs" />
diff --git a/Duplicati/Library/Backend/SSH/Duplicati.Library.Backend.SSH.csproj b/Duplicati/Library/Backend/SSH/Duplicati.Library.Backend.SSH.csproj
index 4c3fae1..111ece8 100644
--- a/Duplicati/Library/Backend/SSH/Duplicati.Library.Backend.SSH.csproj
+++ b/Duplicati/Library/Backend/SSH/Duplicati.Library.Backend.SSH.csproj
@@ -78,7 +78,7 @@
     <Compile Include="SSHUI.cs">
       <SubType>UserControl</SubType>
     </Compile>
-    <Compile Include="SSHUI.designer.cs">
+    <Compile Include="SSHUI.Designer.cs">
       <DependentUpon>SSHUI.cs</DependentUpon>
     </Compile>
     <Compile Include="OutputStreamWrapper.cs" />
diff --git a/Duplicati/Library/Backend/TahoeLAFS/Duplicati.Library.Backend.TahoeLAFS.csproj b/Duplicati/Library/Backend/TahoeLAFS/Duplicati.Library.Backend.TahoeLAFS.csproj
index 1e13958..37d51dc 100644
--- a/Duplicati/Library/Backend/TahoeLAFS/Duplicati.Library.Backend.TahoeLAFS.csproj
+++ b/Duplicati/Library/Backend/TahoeLAFS/Duplicati.Library.Backend.TahoeLAFS.csproj
@@ -113,7 +113,7 @@
     <Compile Include="TahoeUI.cs">
       <SubType>UserControl</SubType>
     </Compile>
-    <Compile Include="TahoeUI.designer.cs">
+    <Compile Include="TahoeUI.Designer.cs">
       <DependentUpon>TahoeUI.cs</DependentUpon>
     </Compile>
   </ItemGroup>
diff --git a/Duplicati/Library/Backend/WEBDAV/Duplicati.Library.Backend.WEBDAV.csproj b/Duplicati/Library/Backend/WEBDAV/Duplicati.Library.Backend.WEBDAV.csproj
index 6ac8c03..9b945e6 100644
--- a/Duplicati/Library/Backend/WEBDAV/Duplicati.Library.Backend.WEBDAV.csproj
+++ b/Duplicati/Library/Backend/WEBDAV/Duplicati.Library.Backend.WEBDAV.csproj
@@ -76,7 +76,7 @@
     <Compile Include="WebDAVUI.cs">
       <SubType>UserControl</SubType>
     </Compile>
-    <Compile Include="WebDAVUI.designer.cs">
+    <Compile Include="WebDAVUI.Designer.cs">
       <DependentUpon>WebDAVUI.cs</DependentUpon>
     </Compile>
   </ItemGroup>
