diff --git a/Cargo.toml b/Cargo.toml
index cdbff82..b527d24 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "news_flash_gtk"
-version = "0.0.0"
+version = "3.0.2"
 authors = ["Jan Lukas Gernert <jangernert@gmail.com>"]
 description = "Feed Reader written in GTK"
 edition = "2018"
@@ -37,8 +37,8 @@ futures = { version = "0.3", features = ["thread-pool"] }
 futures-util = "0.3"
 regex = "1.9"
 feedly_api = "0.5"
-news-flash = { git = "https://gitlab.com/news_flash/news_flash.git" }
-#news-flash = { path = "../news_flash_base/" }
+#news-flash = { git = "https://gitlab.com/news_flash/news_flash.git" }
+news-flash = { path = "../news_flash-1e1ae1d7a8750ba23053de7bef84110bca805725/" }
 clap = { version = "4.4", features = ["derive"] }
 diffus = { version = "0.10", features = [ "derive" ] }
 xmltree = "0.10"
